enterprise/dev/ci/internal/ci/pipeline.go

```diff
@@ -94,13 +94,19 @@ func GeneratePipeline(c Config) (*bk.Pipeline, error) {
 	// PERF: Try to order steps such that slower steps are first.
 	switch c.RunType {
 	case PullRequest:
-		operations = CoreTestOperations(c.ChangedFiles, buildOptions)
+		operations = CoreTestOperations(c.ChangedFiles, CoreTestOperationsOptions{})
+
+		if c.ChangedFiles.affectsClient() {
+			// triggers a slow pipeline, currently only affects web. It's optional so we
+			// set it up separately from CoreTestOperations
+			appendOps(triggerAsync(buildOptions))
+		}
```

Comment:
```suggestion
		if c.ChangedFiles.affectsClient() {
			// triggers a slow pipeline, currently only affects web. It's optional so we
			// set it up separately from CoreTestOperations
			appendOps(triggerAsync(buildOptions))
		}

		appendOps(CoreTestOperations(c.ChangedFiles, CoreTestOperationsOptions{})...)
```

I think `triggerAsync` needs to be added before `CoreTestOperations` is called.

`CoreTestOperations` will wait for all steps to pass, so it means the async pipeline will not actually start until much later.

Caveat: I'm not sure if this code is the best way of doing this, my Go knowledge is limited 😁

Reply:
Good catch, you are right! Thank you 😁